File: DE_1/contestants/bach_do/bai01.py
with open("bai01.inp","r") as fi:
    n=int(fi.readline())
with open("bai01.out","w") as fo:
    s=0
    d=0
    x=n
    while x!=0:
        d+=x%10
        x//=10
    for i in range(1,d//2+1):
        if d==i**2:
            s+=1
            fo.write(str(d))
        break
    if s==0:
        fo.write(str(d%11))
fi.close()
fo.close()

File: DE_1/contestants/nguyen_duy_hung/bai01.cpp

#include <bits/stdc++.h>
using namespace std;
int main() {
    ifstream inp("bai01.inp");
    ofstream out("bai01.out");
    long long n;
    inp >> n;
    vector<bool> hams(162,false);
    for(long long p=1; p*p<=162; p++) {
        hams[p*p]=true;
    }
    int s=0;
    while(n>=1) {
        s+=n%10;
        n=n/10;
    }
    if(hams[s]==true)
        out << s;
    else
        out << s%11;
    out.close();
    return 0;
}
